core/src/main/scala/com/softwaremill/sttp/TryBackend.scala

```diff
@@ -0,0 +1,29 @@
+package com.softwaremill.sttp
+
+import java.net.HttpURLConnection
+
+import scala.util.Try
+
+/** A Backend that safely wraps SttpBackend exceptions in Try's
+  *
+  * @param delegate An SttpBackend which to which this backend forwards all requests
+  * @tparam S The type of streams that are supported by the backend. `Nothing`,
+  *           if streaming requests/responses is not supported by this backend.
+  */
+class TryBackend[-S](delegate: SttpBackend[Id, S]) extends SttpBackend[Try, S] {
+  override def send[T](request: Request[T, S]): Try[Response[T]] =
+    Try(delegate.send(request))
+
+  override def close(): Unit = delegate.close()
+
+  override def responseMonad: MonadError[Try] = TryMonad
+}
+
+object TryBackend {
+  def apply(options: SttpBackendOptions = SttpBackendOptions.Default,
```

Comment:
Thanks a lot for the contribution! Definitely worth including :)

I'm just wondering whether to include the `TryBackend` object - won't it be confusing that `TryBackend` creates a wrapped `HttpURLConnectionBackend`, while `new TryBackend(...)` is a generic wrapper?

It would be a bit more code to instantiate a default `Try`-backed synchronous wrapper, but for the sake of simplicity, `TryBackend(HttpURLConnectionBackend())` wouldn't leave any doubts.

Reply:
You're absolutely correct. Iis a little bit inconsistent isn't it? I was considering making it a TryHttpURLConnectionBackend() but didn't know if that would be too verbose. I'll change it a bit tomorrow.